File: 2406-decode-the-message/2406-decode-the-message.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        
        # Approach 1 Using Hashing
        # decoder = {}
        # alpha = 96
        # output = ""
        
        # for idx, ele in enumerate(key):
        #     if ele not in decoder and ele != " ":
        #         alpha += 1
        #         decoder[ele] = chr(alpha)
                
        # for char in message:
        #     output += decoder[char] if char in decoder else " "
            
        # return (output)

        # Apporach 2 Without using Hashing
        index = []
        output = ''

        for ele in key:
            if ele != ' ' and ele not in index:
                index.append(ele)

        for ele in message:
            if ele != ' ':
                output += chr(index.index(ele)+97)
            else:
                output += ' '

        return output

File: 2406-decode-the-message/2406-decode-the-message.java
class Solution {
    public String decodeMessage(String key, String message) {
        HashMap<Character, Character>decoder = new HashMap<>();
        char alpha = 'a';
        StringBuilder output = new StringBuilder();
        
        for(int i=0; i<key.length();i++) {
            if(decoder.containsKey(key.charAt(i))==false && key.charAt(i) != ' ') {
                decoder.put(key.charAt(i), alpha++);
            }
        }

        for(int i=0; i<message.length(); i++) {
            if(decoder.containsKey(message.charAt(i)))
                output.append(decoder.get(message.charAt(i)));
            else
                output.append(' ');
        }

        return output.toString();
        
    }
}